ddl/ddl_api.go

```diff
@@ -3588,6 +3584,11 @@ func (d *ddl) dropIndex(ctx sessionctx.Context, ti ast.Ident, isPK bool, indexNa
 	}
 
 	indexInfo := t.Meta().FindIndexByName(indexName.L)
+	var isPK bool
+	if indexName.L == strings.ToLower(mysql.PrimaryKeyName) &&
+		(indexInfo == nil || indexInfo.Primary) {
+		isPK = true
+	}
```

Comment:
wrap this check logic to function `IsPK()`

Reply:
This logic is only used here, so I don't think it is necessary to add a new method.

Reply:
`indexName.L == strings.ToLower(mysql.PrimaryKeyName)` is enough?

Reply:
But this problem may occur with indexes that already exist in the old TiDB cluster. I will add some comments here.